src/engine/QueryPlanner.h

```diff
@@ -226,17 +224,27 @@ class QueryPlanner {
   [[nodiscard]] std::vector<QueryPlanner::SubtreePlan> optimize(
       ParsedQuery::GraphPattern* rootPattern);
 
-  void handleSingleVariableCase(
-      const TripleGraph::Node& node, std::function<void(SubtreePlan)> pushPlan,
-      std::function<void(Permutation::Enum)> addIndexScan);
-
-  void handleTwoVariablesCase(
-      const TripleGraph::Node& node,
-      std::function<void(Permutation::Enum)> addIndexScan);
-
-  void handleThreeVariablesCase(
-      const TripleGraph::Node& node,
-      std::function<void(Permutation::Enum)> addIndexScan);
+  // Helper function used by the seedFromOrdinaryTriple function
+  template <typename PushPlanFunction, typename AddedIndexScanFunction>
+  void indexScanSingleVarCase(const TripleGraph::Node& node,
+                              const PushPlanFunction& pushPlan,
+                              const AddedIndexScanFunction& addIndexScan);
+
+  // Helper function used by the seedFromOrdinaryTriple function
+  template <typename AddedIndexScanFunction>
+  void indexScanTwoVarsCase(const TripleGraph::Node& node,
+                            const AddedIndexScanFunction& addIndexScan) const;
+
+  // Helper function used by the seedFromOrdinaryTriple function
+  template <typename AddedIndexScanFunction>
+  void indexScanThreeVarsCase(const TripleGraph::Node& node,
+                              const AddedIndexScanFunction& addIndexScan) const;
+
+  // Helper function used by the seedFromScansAndText function
```

Comment:
Make this the first of those functions (the other ones are its impls)
and make the comment.

// Add all the possible index scans for the triple represented by the `node`. The triple is "ordinary" in the sense that it is neither a text triple with `ql:contains-word` nor a special pattern trick triple.